BaseStation/MainWindow/MapTileDownloader.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Net;
+using System.IO;
+using System.Collections.Generic;
+using HuskyRobotics.Utilities;
+
+// make sure there is a folder named MapTiles in the working directory
+// google API key is stored in environment variables
+// https://msdn.microsoft.com/en-us/library/bb259689.aspx#Map map system (its bing but google
+// uses the same system)
+// https://google-developers.appspot.com/maps/documentation/static-maps/intro#Zoomlevels google
+// tile download documentation
+
+namespace HuskyRobotics.UI
+{
+    public static class MapTileDownloadManager
+    {
+        // a class to hold the configuration for the map dowload, formated as specified by google
+        public class Configuration
+        {
+            private Dictionary<String, String> parameters;
+
+            public Configuration(int imgWidth, int imgHeight, int scale, int zoom, Tuple<double, double> coords)
+            {
+                parameters = new Dictionary<String, String>();
+
+                SetLocation(coords);
+                SetZoom(zoom);
+                SetImageSize(imgHeight, imgWidth);
+                SetImageScale(scale);
+                parameters.Add("maptype", "satellite");
+            }
+
+            // adds the image dimensions to the current configuration
+            public void SetImageSize(int width, int height)
+            {
+                if (parameters.ContainsKey("size")) parameters.Remove("size");
+                parameters.Add("size", width + "x" + height);
+            }
+
+            // returns the dimensions of the image in the configuration
+            public String GetImageDim()
+            {
+                String size;
+                parameters.TryGetValue("size", out size);
+                return size;
+            }
+
+            // adds the scale to the current configuration
+            public void SetImageScale(int scale)
+            {
+                if (parameters.ContainsKey("scale")) parameters.Remove("scale");
+                parameters.Add("scale", scale.ToString());
+            }
+
+            // returns the scale of the image in the configuration
+            public String GetScale()
+            {
+                String scale;
```

Comment:
Give this a reasonable default (instead of null), or throw an error when the scale isn't set. (A default is probably a better option in this case)

This applies to a few other places in the code as well.